app/components/layout/sidebar/index.tsx

```diff
@@ -48,68 +46,114 @@ export default function Sidebar({ user }: Props) {
+    <>
+      {/* this component is named sidebar as of now but also serves as a mobile navigation header in mobile device */}
+      <header
+        id="header"
+        className="flex items-center justify-between border-b bg-white p-4 md:hidden"
+      >
+        <Link to="." title="Home" className="block h-[32px]">
           <img
-            src="/images/shelf-symbol.png"
-            alt="Shelf Logo"
-            className="mx-1.5 inline h-[32px]"
+            src="/images/logo-full-color(x2).png"
+            alt="logo"
+            className="h-full"
           />
-          <span className="logo-text transition duration-200 ease-linear">
-            <ShelfTypography />
-          </span>
         </Link>
-        <button
-          className="transition-all duration-200 ease-linear"
-          onClick={manageUncollapsedSidebar}
-        >
-          <i className="icon text-gray-500">
-            {maintainUncollapsedSidebar ? <ActiveSwitchIcon /> : <SwitchIcon />}
-          </i>
+        <button className="menu-btn relative z-50" onClick={toggleSidebar}>
+          <span
+            className={tw(
+              "mb-1 block h-[2px] w-[19px] rounded-full bg-gray-500 transition-all",
+              isSidebarCollapsed ? "" : "translate-y-[6px] rotate-45"
+            )}
+          ></span>
+          <span
+            className={tw(
+              "mb-1 block h-[2px] w-[14px] rounded-full bg-gray-500 transition-all",
+              isSidebarCollapsed ? "opacity-1" : "invisible opacity-0"
+            )}
+          ></span>
+          <span
+            className={tw(
+              "mb-1 block h-[2px] w-[19px] rounded-full bg-gray-500 transition-all",
+              isSidebarCollapsed ? "" : "translate-y-[-6px] -rotate-45"
+            )}
+          ></span>
         </button>
-      </div>
-      <div className="flex-1">
-        <ul className="menu mt-10">
-          {menuItems.map((item) => (
-            <li key={item.label}>
-              <NavLink
-                className={({ isActive }) =>
-                  tw(
-                    "my-1 flex items-center gap-3 rounded-md px-3 py-2.5 text-text-md font-semibold text-gray-700 transition-all duration-75 hover:bg-gray-100 hover:text-gray-900",
-                    isActive ? "bg-gray-100 text-gray-900" : ""
-                  )
-                }
-                to={item.to}
-                data-test-id={`${item.label.toLowerCase()}SidebarMenuItem`}
-              >
-                <i className="icon text-gray-500">{item.icon}</i>
-                <span className="text whitespace-nowrap transition duration-200 ease-linear">
-                  {item.label}
-                </span>
-              </NavLink>
-            </li>
-          ))}
-        </ul>
-      </div>
-      <div className="mt-auto">
-        <SidebarBottom
-          user={{
-            ...user,
-          }}
-        />
-      </div>
-    </aside>
+      </header>
+      <div
```

Comment:
This works really good but its super hard to read like this, as its difficult to figure which html is which element. Can you please place this in a separate component.

Reply:
@DonKoko do you want me to convert this overlay into a component or the whole header?

Reply:
The overlay